### PR TITLE
BL-1053-as-a-case-manager-when-i-enter-a-case-note-i-want-the-time-and-date-to-be-stamped-within-the-note-so-that-i-can-have-a-record-of-the-time-date-that-a-case-note-was-entered

### PR DESCRIPTION
**Pull request by Kobramall**
loom videos:
      (https://www.loom.com/share/7719c3161b454887b4cfb6aa91ba88b1)

https://www.loom.com/share/9cfa606b192649e5bc2452e581068829

https://www.loom.com/share/3ab48d957c464f45b36f512408f7b38d

  When notes are saved there is a timestamp of the date when it was saved displayed after the notes text. I did this by:
    1.add a function that fetched the client's notes from the backend
    2. set the notes the result 
    3. fixed the code to make the timestamp works both when the website is rendered and when a note is saved
    4. I add a post of the new note to the backend

### CODE REVIEW

File: src/components/common/CaseDetails/CaseDetails.js [hunk 1]
```diff
@@ -28,6 +28,11 @@ function CaseDetails() {
   const [filterSearch, setFilterSearch] = useState('');
   const [filteredResults, setFilteredResults] = useState([]);
 
+  //State slice for notes
+  const [notes, setNotes] = useState([
```
Comment (ashtilawat23) [inline src/components/common/CaseDetails/CaseDetails.js:32]: Can we import in the notes from the backend using GET `/:clientId/notes` next?

File: src/components/common/CaseDetails/CaseDetails.js [hunk 2]
```diff
@@ -188,28 +201,15 @@ function CaseDetails() {
...
-                    <div>
-                      <h3>Title here2</h3>
-                      <p>Note goes here.2</p>
-                    </div>
-                    <div>
-                      <h3>Title here2</h3>
-                      <p>Note goes here.2</p>
+                      <p>
+                        {item.note}
+                        <span>{item.date}</span>
```
Comment (ashtilawat23) [inline src/components/common/CaseDetails/CaseDetails.js:211]: Well done.

**Review by ashtilawat23 — CHANGES_REQUESTED** [review]
Changes requested: 

- Remove package-lock.json from the PR. 
- If you want to add the `yaml` package, please do so in a separate PR so we can manage dependencies properly. 
- Let's get this connected to the backend

File: src/components/common/CaseDetails/CaseDetails.js (hunk 1 above)
Comment (Kobramall) [reply src/components/common/CaseDetails/CaseDetails.js:32]: For sure!

File: src/components/common/CaseDetails/CaseDetails.js [hunk 3]
```diff
@@ -188,28 +233,20 @@ function CaseDetails() {
                   </div>
                 );
               })
-            : placeholderArray.map(item => {
+            : notes.map(item => {
+                let date = new Date(item.created_at);
```
Comment (JordonP123) [inline src/components/common/CaseDetails/CaseDetails.js:237]: I see you are grabbing the month/date/year.  Did that want a time when this was added?  If so, you can format the date using new Date(item.created_at).toLocaleString()

Comment (JordonP123) [reply src/components/common/CaseDetails/CaseDetails.js:237]: If not you could still clean it up a little bit using toLocaleDateString() instead

File: src/components/common/CaseDetails/CaseDetails.js [hunk 4]
```diff
@@ -49,12 +66,40 @@ function CaseDetails() {
...
+    //Create a time stamp in MM/DD/YYYY fomate
+    let timeStamp = Date.now();
+    new Intl.DateTimeFormat('en-US', {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+    }).format(timeStamp);
```
Comment (JordonP123) [inline src/components/common/CaseDetails/CaseDetails.js:78]: Do you use this timestamp anywhere?  I can't find it.

**Review by JordonP123 — COMMENTED** [review]
Overall, looks solid!  The timestamp I don't think is being used, and the date is only recording the MM/DD/YYYY.  Not sure if they want the time!

File: src/components/common/CaseDetails/CaseDetails.js (hunk 4 above)
Comment (Kobramall) [reply src/components/common/CaseDetails/CaseDetails.js:78]: Yeah I meant to delete that but guess I messed it

**Pull request merged**